test/provisioner/provision_azure.go

```diff
@@ -8,37 +8,351 @@ package provisioner
 import (
 	"context"
 
+	"errors"
+	"fmt"
+	"os"
 	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/containerservice/mgmt/containerservice"
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/network/mgmt/network"
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/resources/mgmt/resources"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/to"
+	log "github.com/sirupsen/logrus"
+	kconf "sigs.k8s.io/e2e-framework/klient/conf"
 )
 
 func init() {
 	newProvisionerFunctions["azure"] = NewAzureCloudProvisioner
 }
 
-// AzureCloudProvisioner implements the CloudProvisioner interface for Azure.
+func CreateResourceGroup() error {
```

Comment:
Any reason declare this function as public, will it be used out side of this package?

Reply:
Not intended to be public, changed the name.